v2/bigquery-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SpannerConverters.java

```diff
@@ -0,0 +1,256 @@
+                        instant.get(DateTimeFieldType.year()),
+                        instant.get(DateTimeFieldType.monthOfYear()),
+                        instant.get(DateTimeFieldType.dayOfMonth())));
+              };
+            }
+            throw new UnsupportedOperationException(
+                String.format(
+                    "Converting logical type %s for spanner mutation not implemented, baseType=%s",
+                    logicalType.getIdentifier(), logicalType.getBaseType().getTypeName()));
+
+          case BYTE:
+          case MAP:
+          case DECIMAL:
+          default:
+            throw new UnsupportedOperationException(
+                String.format("Converting type %s for spanner mutation not implemented",
+                    type.getTypeName()));
+        }
+      }
+
+      private SerializableFunction<List<?>, Value> getArrayValueGetter(FieldType type) {
+        Preconditions.checkNotNull(type.getCollectionElementType());
+        switch (type.getCollectionElementType().getTypeName()) {
+          case BOOLEAN:
+            return (List<?> lst) -> Value
+                .boolArray(lst.stream().map(o -> (Boolean) o).collect(toList()));
+          case BYTES:
+            return (List<?> lst) -> Value.bytesArray(
+                lst.stream().map(o -> ByteArray.copyFrom((byte[]) o)).collect(toList()));
+          case DATETIME:
+            return (List<?> lst) -> Value.timestampArray(
+                lst.stream().map(o -> com.google.cloud.Timestamp
+                    .ofTimeMicroseconds(((AbstractInstant) o).getMillis() * 1000))
+                    .collect(toList()));
+          case FLOAT:
+            return (List<?> lst) -> Value
+                .float64Array(lst.stream().map(o -> Double.valueOf((Float) o)).collect(toList()));
+          case DOUBLE:
+            return (List<?> lst) -> Value
+                .float64Array(lst.stream().map(o -> (Double) o).collect(toList()));
+          case INT16:
+            return (List<?> lst) -> Value
+                .int64Array(lst.stream().map(o -> Long.valueOf((Short) o)).collect(toList()));
+          case INT32:
+            return (List<?> lst) -> Value
+                .int64Array(lst.stream().map(o -> Long.valueOf((Integer) o)).collect(toList()));
+          case INT64:
+            return (List<?> lst) -> Value
+                .int64Array(lst.stream().map(o -> (Long) o).collect(toList()));
+          case STRING:
+            return (List<?> lst) -> Value
+                .stringArray(lst.stream().map(Object::toString).collect(toList()));
+          case ARRAY:
+            throw new UnsupportedOperationException(
+                "Cannot construct array with element type ARRAY<?> because nested arrays are not supported by Spanner");
+          case BYTE:
+          case LOGICAL_TYPE:
+          case MAP:
+          case ROW:
+          case DECIMAL:
+          default:
+            throw new UnsupportedOperationException(
```

Comment:
Conversion map needs to be documented somewhere. Add to readme what is supported and how it's converted (eg datetime->Timestamp , int64-> long)